packages/cms/src/components/Profile.jsx

```diff
@@ -92,7 +93,14 @@ class Profile extends Component {
   }
 
   render() {
-    const {profile, loading} = this.state;
+    const {isIE, profile, loading} = this.state;
+
+    // IE check needed for position: sticky fallback
+    if (typeof window !== "undefined") {
+      if (/*@cc_on!@*/false || !!document.documentMode) { // eslint-disable-line spaced-comment
+        this.setState({isIE: true});
+      }
+    }
```

Comment:
using `setState` within the render function is a no-no... can this logic just be moved right into the constructor?

Reply:
Ah okay. Because we'd be needlessly updating the state every time we need to re-render the component?

Reply:
https://github.com/facebook/react/issues/5342

Reply:
Apparently setState in the constructor is also a no-no. How about moving it to a `componentDidMount()` lifecycle method?